metricbeat/module/kibana/rules/data.go

```
@@ -0,0 +1,124 @@
+		"status":            c.Str("status"),
+	}
+
+	rulesSchema = s.Schema{
+		"overdue": c.Dict("overdue", s.Schema{
+			"count": c.Int("count"),
+			"duration": c.Dict("duration", s.Schema{
+				"p50": c.Float("p50"),
+				"p99": c.Float("p99"),
+			}),
+		}),
+		"failures":   c.Int("failures"),
+		"executions": c.Int("executions"),
+	}
+)
+
+type response struct {
+	Rules       map[string]interface{} `json:"rules"`
+	Kibana      map[string]interface{} `json:"kibana"`
+	ClusterUuid string                 `json:"cluster_uuid"`
+}
+
+func eventMapping(r mb.ReporterV2, content []byte, isXpack bool) error {
+	var data response
+	err := json.Unmarshal(content, &data)
+	if err != nil {
+		return errors.Wrap(err, "failure parsing Kibana Rules API response")
+	}
+
+	event := mb.Event{ModuleFields: common.MapStr{}, RootFields: common.MapStr{}}
+
+	// Set elasticsearch cluster id
+	event.ModuleFields.Put("elasticsearch.cluster.id", data.ClusterUuid)
+
+	kibana, _ := kibanaSchema.Apply(data.Kibana)
+	if err != nil {
+		event.Error = elastic.MakeErrorForMissingField("kibana", elastic.Kibana)
+		return event.Error
+	}
+
+	// Set service ID
+	serviceId, err := kibana.GetValue("uuid")
+	if err != nil {
+		event.Error = elastic.MakeErrorForMissingField("kibana.uuid", elastic.Kibana)
+		return event.Error
+	}
+	event.RootFields.Put("service.id", serviceId)
+
+	// Set service version
+	version, err := kibana.GetValue("version")
+	if err != nil {
+		event.Error = elastic.MakeErrorForMissingField("kibana.version", elastic.Kibana)
+		return event.Error
+	}
+	event.RootFields.Put("service.version", version)
+
+	// Set service address
+	serviceAddress, err := kibana.GetValue("transport_address")
+	if err != nil {
+		event.Error = elastic.MakeErrorForMissingField("kibana.transport_address", elastic.Kibana)
+		return event.Error
+	}
+	event.RootFields.Put("service.address", serviceAddress)
+
+	rulesFields, err := rulesSchema.Apply(data.Rules)
+	if err != nil {
```

You probably found some example where we handle those errors. But I think that in general it's better to ignore them. It really depends if you want "partial data" or an "all or nothing". If one single field is missing, you get nothing, even if the rest of the fields are there. That's why it might be worth to ignore the error.